### PR TITLE
Improve advanced select results

### LINKED ISSUE #732: Fix PostTypeRecordsSelect search results
### Description
The component `PostTypeRecordsSelect` is not working properly when the result set has more than 10 posts.

### Steps to reproduce
1. Have a site with 25+ posts (with similar words in the title)
2. Add a query loop
3. Add "Include posts" and search for a post that is not showing initially in the options.

#### Expected behavior
The search result brings all posts correctly

#### Actual behavior
The search result is only adding the first 10 posts from the result set.

### PR DESCRIPTION
**Pull request by JeanPaiva**
close #732

### CODE REVIEW

**Comment by tomusborne** [comment]
@JeanPaiva Do we need to include this is the post select when querying/searching for specific posts in the Dynamic Content area?

**Comment by JeanPaiva** [comment]
@tomusborne the posts select component is the same for both, the only thing missing was the label on the dynamic content.

**Comment by tomusborne** [comment]
Playing with this. I think it may be better to:

1. Allow search by ID when using `#` or if the search is a number only.
2. Change the help text to: `Search by name or ID.`

What do you think?

**Comment by JeanPaiva** [comment]
@tomusborne changed to accept only numbers as ID search and removed the # tag search.

File: src/blocks/query-loop/components/inspector-controls/controls/TaxonomyParameterControl.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ export default function TaxonomyParameterControl( props ) {
 
 							setTerms( newTerms );
 						} }
-						help={ terms.length === 0 ? __( 'You must select at least one term.', 'generateblocks' ) : '' }
+						help={ terms.length === 0 ? __( 'You must select at least one term. Use #123 with your ID to search by term ID.', 'generateblocks' ) : help }
```
Comment (tomusborne) [inline src/blocks/query-loop/components/inspector-controls/controls/TaxonomyParameterControl.js:83]: @JeanPaiva Need to update help here.

**Pull request merged**